src/npm-publish.ts

```diff
@@ -25,7 +25,7 @@ export async function npmPublish(opts: Options = {}): Promise<Results> {
 
   let results: Results = {
     package: manifest.name,
-    type: diff || "none",
+    type: options.dryRun ? "dry-run" : (diff || "none"),
```

Comment:
Instead of overloading the `type` property, let's add a new property to the `Results` object (i.e. `dryRun: boolean`).  That way, scripts can still check the `type` property like normal, regardless of whether it was a dry run or not.